README.md

```diff
@@ -1,10 +1,54 @@
-# Website
+# Project Layout
+
+### The project is broken down into 4 sections
+- Concepts - General Uniswap information or concepts useful for using Uniswap products, such as *Liquidity* and *Fees*
+- Contracts - The Uniswap smart contracts such as *Permit2*, or groupings such as the V3 Contracts
+- SDKs - The Uniswap SDKs such the *v3-sdk*
+- APIs - The Uniswap APIs such the *Subgraph API*
+
+### Each section contains items of that category
+For each item there should be:
+- *Overview*
+- *Guides*
+- *Technical Reference*
+
+## Deep dive into the requirements of each part
+
+### Overview
+This should be an overview of the product.
+It should address points such as:
+
+- What are the high level components of the product
+- What are the high level functionalities
+- Where does the source code of the product live
+- Where does the code artifact live (eg *npm*) and how to integrate with it
+
+A good example is the [V3 Smart Contracts](./docs/contracts/v3/overview.md).
+
+### Guides
+This should contain guides for using the product.
+Each guide must have 3 parts:
+- An introduction
+- A step-by-step walk through of each line of code. The guide should not include code that is not part of the example, but should instead reference or link those pieces.
```

Comment:
It feels a bit weird to ask for a step-by-step walkthrough but not include any code. I think there should be an expectation we don't post large blocks of code, but I think we should leave room for singular lines, data structures, etc

```suggestion
- A walkthrough of the provided example code. The guide should not directly include large blocks of code, but should instead reference/link to this code as needed, including snippets sparingly when required.
```